src/dk/aau/cs/ds306e18/tournament/rlbot/configuration/ConfigFile.java

```diff
@@ -0,0 +1,280 @@
+package dk.aau.cs.ds306e18.tournament.rlbot.configuration;
+
+import java.io.*;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Abstraction of a '.cfg'-file.
+ */
+public class ConfigFile {
+
+    // Can parse files like this: https://docs.python.org/3/library/configparser.html#supported-ini-file-structure
+    // Inspired by: https://stackoverflow.com/a/15638381
+
+    private final Pattern sectionPattern = Pattern.compile("\\[([^\\[\\]]*)\\]\\s*");
+    private final Pattern keyValuePattern = Pattern.compile("([^=:]*)(=|:)(.*)");
+    private final Pattern multiLineValuePattern = Pattern.compile("[^\\S\\r\\n]+(.*)");
+    private final Pattern commentAndEmptyPattern = Pattern.compile("\\s*[#;].*|\\s*");
+
+    private Map<String, Map<String, String>> entries = new HashMap<>();
+
+    /**
+     * Create empty ConfigFile.
+     */
+    public ConfigFile() {
+
+    }
+
+    /**
+     * Create ConfigFile based on a the entries of a config file.
+     * @param file the file to parse.
+     * @throws IOException
+     */
+    public ConfigFile(File file) throws IOException {
+        load(file);
+    }
+
+    /**
+     * Fills this ConfigFile with the entries from the given config file.
+     * @param file the config file to parse.
+     * @throws IOException
+     */
+    public void load(File file) throws IOException {
+        try (BufferedReader br = new BufferedReader(new FileReader(file))) {
+            int lineNumber = 0;
+            String line;
+            Map<String, String> currentSectionMap = null;
+            String key = null;
+            String value = null;
+            Matcher m;
+            while ((line = br.readLine()) != null) {
```

Comment:
Is this xv6-inspired? :eyes:

Reply:
Just resolve when finished chuckling.

Reply:
Not at all. Just a very standard way to parsing lines in a file ;)